File: app/src/types/note.ts
export type Note = {
  id: string;
  paperId: string;
  page: number;
  x: number;
  y: number;
  content: string;
  color?: string;
  createdAt: string;
  updatedAt: string;
};

export type NewNote = {
  paperId: string;
  page: number;
  x: number;
  y: number;
  content: string;
  color?: string;
};

export type UpdateNote = {
  id: string;
  content?: string;
  color?: string;
};

File: src-tauri/src/domain/note.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct Note {
    pub id: String,
    pub paper_id: String,
    pub page: i32,
    pub x: f32,
    pub y: f32,
    pub content: String,
    pub color: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct NewNote {
    pub paper_id: String,
    pub page: i32,
    pub x: f32,
    pub y: f32,
    pub content: String,
    pub color: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct UpdateNote {
    pub id: String,
    pub content: Option<String>,
    pub color: Option<String>,
}
